lambdas/service/app.py

```diff
@@ -1752,6 +1746,36 @@ def validate_wait(wait: Optional[str]) -> Optional[int]:
                                                    headers=headers)
 
 
+@app.route('/repository/sources', methods=['GET'], cors=True, method_spec={
+    'summary': 'List available data sources for the selected catalog',
+    'tags': ['Repository'],
+    'parameters': [catalog_param_spec],
+    'responses': {
+        '200': {
+            'description': format_description('''
+                List the sources in the underlying data repository that the
+                current authenticated user is authorized to access.
+            '''),
+            **responses.json_content(
+                schema.object(sources=array_of_object_spec)
+            )
+        },
+        # FIXME Determine public snapshots
+        #       https://github.com/DataBiosphere/azul/issues/2978
+        '401': {
+            'description': format_description('''
+                This endpoint requires authentication.
+            ''')
+        }
+    }
+})
+def repository_sources() -> Response:
+    validate_params(app.current_request.query_params, catalog=validate_catalog)
+    sources = app.repository_controller.accessible_sources(app.catalog,
+                                                           app.current_request.authentication)
```

Comment:
Let the controller unpack the request. At some point we should move `app.catalog` to `app.current_request.catalog`

```suggestion
                                                           app.current_request)
```